File: ccif/src/java/other/org/himalay/ccif/PurchasePin.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:08 EST 2016")

public  class PurchasePin extends BinStruct implements PublicBinMsg {




    // members variables
    // purchagePinList
    public ByteArray purchagePinList ;

    public PurchasePin () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize purchagePinList
        purchagePinList= new ByteArray();
        purchagePinList.setSizeType("FIRST_UI16");
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read purchagePinList
        {
            retVal+=purchagePinList.read(istream);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for purchagePinList**/
        }

        // write purchagePinList
        {
            retVal += purchagePinList.write(ostream);
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("PurchasePin\n");
        dc.increaseIndent();
        int retVal= 0;
        // write purchagePinList
        dc.indent();
        dc.getPs().print("purchagePinList: "+purchagePinList.getSize()+"(0x"+Integer.toHexString(purchagePinList.getSize())+")\n");
        this.purchagePinList.dump(dc);
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for purchagePinList
    //public ByteArray getPurchagePinList()
    //{
    //    return purchagePinList ;
    //}


    // Setter for purchagePinList
    //public void setPurchagePinList(ByteArray val)
    //{
    //    this.purchagePinList= val;
    //}


    public void setPurchagePinList(byte[] val)
    {
        this.purchagePinList.setData(val);
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/PurchasePin.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "PurchasePin.h"

#ifdef org_himalay_ccif_PurchasePin__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
PurchasePin::PurchasePin() {
    references= 0;


    // purchagePinList
    org_himalay_msgs_runtime_ByteArray_NEW(purchagePinList);
    purchagePinList->setSizeType("FIRST_UI16");
}
// Destructor
PurchasePin::~PurchasePin() {
    // purchagePinList
    org_himalay_msgs_runtime_ByteArray_DELETE( purchagePinList);
}

int PurchasePin::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // purchagePinList
    {
        retVal+=purchagePinList->read(istream);
    }
    return retVal;
}

int PurchasePin::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read purchagePinList
    {
        retVal+=purchagePinList->read(istream);
    }

    return retVal;
}

int PurchasePin::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for purchagePinList**/
    }

    int retVal= 0;
    // write purchagePinList
    {
        retVal += purchagePinList->write(ostream);
    }

    return retVal;
}

int PurchasePin::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("PurchasePin\n");
    dc.increaseIndent();
    int retVal= 0;
    // write purchagePinList
    dc.indent();
    dc.getPs().print("purchagePinList");
    purchagePinList->dump(dc);
    dc.decreaseIndent();
    return retVal;
}


// Getter for purchagePinList
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE PurchasePin::getPurchagePinList()
//{
//return this->purchagePinList;
//}

// Setter for purchagePinList
//void PurchasePin::setPurchagePinList(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->purchagePinList= val;
//}

int PurchasePin::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_PurchasePin__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::PurchasePin* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::PurchasePin* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code